cmd/mode.go

```diff
@@ -28,27 +26,18 @@ type Mode struct {
 
 // Build assets.
 func (r *Mode) Build(application *api.Application) (err error) {
-	binDir := path.Join(BinDir, "maven")
-	maven := &repository.Maven{
-		M2Dir:  M2Dir,
-		BinDir: binDir,
-		Remote: repository.Remote{
-			Identities: application.Identities,
-		},
-	}
 	if !r.Binary {
 		err = r.fetchRepository(application)
 		return
 	}
-
 	if r.Artifact != "" {
 		err = r.getArtifact()
 		return
 	}
-
-	if application.Binary != "" {
-		err = r.mavenArtifact(application, maven)
-		return
+	if strings.Contains(application.Binary, ":") {
+		r.path.binary = "mvn://" + application.Binary + "@" + BinDir
+	} else {
+		r.path.binary = BinDir
```

Comment:
Ideally, the UI would store the coordinates in this format but probably cannot be done in time for 0.5.